api/src/main/java/net/kyori/adventure/key/Key.java

```diff
@@ -29,7 +29,26 @@
 import org.checkerframework.checker.nullness.qual.NonNull;
 
 /**
- * A key.
+ * An identifying object used to fetch and/or store unique objects.
+ *
+ * <p>A key consists of:</p>
+ * <ul>
+ *   <li> a namespace - in most cases this should be your plugin or organization name</li>
+ *   <li> a value - what this key leads to, e.g "translations" or "entity.firework_rocket.blast"</li>
+ * </ul>
+ *
+ * <p>Valid characters for namespaces and keys are [a-z0-9_.-].</p>
+ *
+ * <p>Some examples of possible custom keys:</p>
+ * <ul>
+ *   <li> myPlugin:translations</li>
```

Comment:
No capital letters are allowed.

Reply:
Yes, I've fallen in this trap once earlier..